nucliadb/nucliadb/train/generators/image_classifier.py

```diff
@@ -0,0 +1,186 @@
+    shard_replica_id: str,
+) -> AsyncIterator[ImageClassificationBatch]:
+    request = StreamRequest()
+    request.shard_id.id = shard_replica_id
+    request.reload = True
+    batch = ImageClassificationBatch()
+    async for item in node.stream_get_fields(request):
+        rid = item.uuid
+        resource = await get_resource_from_cache(kbid, rid)
+        if resource is None:
+            logger.error(f"Resource {rid} does not exist on DB")
+            return
+
+        _, field_type, field_key = item.field.split("/")
+        field_type = KB_REVERSE[field_type]
+
+        if field_type not in VISUALLY_ANNOTABLE_FIELDS:
+            continue
+
+        page_selections = await get_page_selections(resource, field_key)
+        if len(page_selections) == 0:
+            # Generating a payload without annotations makes no sense
+            continue
+
+        field = await resource.get_field(field_key, field_type, load=True)
+        page_structure = await get_page_structure(field)
+
+        print("LEN DE PAGE STRCTURE:", len(page_structure))
+        for page, (page_uri, ps) in enumerate(page_structure):
+            pawls_payload = {
+                "width": ps.page.width,
+                "height": ps.page.height,
+                "tokens": [
+                    {
+                        "x": token.x,
+                        "y": token.y,
+                        "width": token.width,
+                        "height": token.height,
+                        "text": token.text,
+                        "line": token.line,
+                    }
+                    for token in ps.tokens
+                ],
+                "annotations": [
+                    {
+                        "page": page,
+                        "label": {
+                            "text": selection.label,
+                        },
+                        "bounds": {
+                            "top": selection.top,
+                            "left": selection.left,
+                            "right": selection.right,
+                            "bottom": selection.bottom,
+                        },
+                        "tokens": [
+                            {
+                                "pageIndex": page,
+                                "tokenIndex": token_id,
+                            }
+                            for token_id in selection.token_ids
+                        ],
+                    }
+                    for selection in page_selections[page]
+                ],
+            }
+
+            ic = ImageClassification()
+            ic.uri = page_uri
+            ic.selections = json.dumps(pawls_payload)
+            batch.data.append(ic)
+
+            if len(batch.data) == trainset.batch_size:
+                yield batch
+                batch = ImageClassificationBatch()
+
+    if len(batch.data):
+        yield batch
+
+
+async def get_page_selections(
+    resource: Resource, field_key: str
```

Comment:
Is field key the field id only? If so, I think it should include both field type and field id, as you can have the same field id in different field types.

Reply:
Changed for `field` and properly checked field id and type